sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeWriter.java

```diff
@@ -134,13 +135,17 @@ public final void write(int ordinal, CalendarInterval input) {
     // grow the global buffer before writing data.
     grow(16);
 
-    // Write the months, days and microseconds fields of Interval to the variable length portion.
-    Platform.putInt(getBuffer(), cursor(), input.months);
-    Platform.putInt(getBuffer(), cursor() + 4, input.days);
-    Platform.putLong(getBuffer(), cursor() + 8, input.microseconds);
-
-    setOffsetAndSize(ordinal, 16);
+    if (input == null) {
+      BitSetMethods.set(getBuffer(), startingOffset, ordinal);
+      setOffsetAndSize(ordinal, 0);
```

Comment:
to match `UnsafeRow.setInterval`, shall we use 16 as size here?

Reply:
yea